openpype/modules/deadline/plugins/publish/submit_3dmax_deadline.py

```diff
@@ -0,0 +1,237 @@
+                "UserName": deadline_user,
+
+                "Plugin": instance.data["plugin"],
+                "Group": self.group,
+                "Pool": self.deadline_pool,
+                "secondaryPool": self.deadline_pool_secondary,
+                "Frames": frames,
+                "ChunkSize": self.chunk_size,
+                "Priority": instance.data.get("priority", self.priority),
+                "Comment": comment,
+                "FramesPerTask": self.framePerTask
+            },
+            "PluginInfo": {
+                # Input
+                "SceneFile": filepath,
+                "Version": "2023",
+                "SaveFile": True,
+                # Mandatory for Deadline
+                # Houdini version without patch number
+
+                "IgnoreInputs": True
+            },
+
+            # Mandatory for Deadline, may be empty
+            "AuxFiles": []
+        }
+        # Include critical environment variables with submission + api.Session
+        keys = [
+            # Submit along the current Avalon tool setup that we launched
+            # this application with so the Render Slave can build its own
+            # similar environment using it, e.g. "maya2018;vray4.x;yeti3.1.9"
+            "AVALON_TOOLS",
+            "OPENPYPE_VERSION"
+        ]
+        # Add mongo url if it's enabled
+        if context.data.get("deadlinePassMongoUrl"):
+            keys.append("OPENPYPE_MONGO")
+
+        environment = dict({key: os.environ[key] for key in keys
+                            if key in os.environ}, **legacy_io.Session)
+
+        payload["JobInfo"].update({
+            "EnvironmentKeyValue%d" % index: "{key}={value}".format(
+                key=key,
+                value=environment[key]
+            ) for index, key in enumerate(environment)
+        })
+
+        # Include OutputFilename entries
+        # The first entry also enables double-click to preview rendered
+        # frames from Deadline Monitor
+        output_data = {}
+        # need to be fixed
+        for i, filepath in enumerate(instance.data["expectedFiles"]):
+            dirname = os.path.dirname(filepath)
+            fname = os.path.basename(filepath)
+            output_data["OutputDirectory%d" % i] = dirname.replace("\\", "/")
+            output_data["OutputFilename%d" % i] = fname
+
+        if not os.path.exists(dirname):
+            self.log.info("Ensuring output directory exists: %s" %
+                          dirname)
+            os.makedirs(dirname)
+
+        plugin_data = {}
+        project_setting = get_project_settings(
+            legacy_io.Session["AVALON_PROJECT"]
+        )
+
+        multipass = get_multipass_setting(project_setting)
+        if multipass:
+            plugin_data["DisableMultipass"] = 0
+        else:
+            plugin_data["DisableMultipass"] = 1
+
+        if self.use_published:
+            old_output_dir = os.path.dirname(expected_files[0])
```

Comment:
beware that `expected_files` might be None here. The code above will work with it somehow, but here it will crash. Of course there should always be `expected_files`, but maybe we should error out early if this is not so, probably on the line 74:

```python
expected_files = instance.data.get("expectedFiles")
```
it should either bail out on AttributeError or with explicit check and informative error report (preferred)